src/vue/forms/LoginForm.spec.js

```diff
@@ -0,0 +1,188 @@
+      for (const fieldValue of fieldValues.valid) {
+        it(`considers ${fieldValue} a valid ${fieldName}`, () => {
+          expect(TestHelper.isFieldValid(
+            wrapper,
+            fieldName,
+            fieldValue
+          ))
+            .to.be.true
+        })
+      }
+      for (const fieldValue of fieldValues.invalid) {
+        it(`considers ${fieldValue} an invalid ${fieldName}`, () => {
+          expect(TestHelper.isFieldValid(
+            wrapper,
+            fieldName,
+            fieldValue
+          ))
+            .to.be.false
+        })
+      }
+    }
+
+    const fieldBindings = {
+      '#login-email': 'form.email',
+      '#login-password': 'form.password'
+    }
+
+    for (const [selector, model] of Object.entries(fieldBindings)) {
+      it(`$v.${model} is touched after blur event emitted on ${selector}`, async () => {
+        const touchField = sinon.spy()
+
+        wrapper.setMethods({ touchField })
+        wrapper.find(selector).vm.$emit('blur')
+
+        expect(touchField.withArgs(model).calledOnce).to.be.true
+      })
+    }
+  })
+
+  describe('methods', () => {
+    let spies
+    let wrapper
+    let sdk = {
+      api: {
+        users: {}
+      }
+    }
+
+    beforeEach(() => {
+      spies = {
+        loadWallet: sinon.spy()
+      }
+
+      const store = new Vuex.Store({
+        modules: {
+          'new-wallet': {
+            namespaced: true,
+            actions: {
+              [vuexTypes.LOAD_WALLET]: spies.loadWallet
+            },
+            getters: {
+              [vuexTypes.wallet]: sinon.stub().returns(new Wallet(
+                'foo@mail.com',
+                'SCKPYOBGIQ5CM3NEZYM25H5SIWZONRUHNMXZ47JBLM6RMBGDIY5NLERG',
+                'GBYJOGI4T7RHIOJFXXSCMOQYXC5BD63Z4FVCUAWZ27UQ4CQPZGG432T6',
+                '4aadcd4eb44bb845d828c45dbd68d5d1196c3a182b08cd22f05c56fcf15b153c'
+              ))
+            }
+          }
+        }
+      })
+
+      sdk.api.users = mockHelper.getApiResourcePrototype('users')
+      sdk.api.users.create = sinon.spy()
+
+      wrapper = shallowMount(LoginForm, {
+        store,
+        localVue,
+        methods: {
+          isFormValid: sinon.stub().returns(true),
+          _doLegacyStuff: sinon.stub().resolves()
+        }
+      })
+    })
+
+    it('submit() loads wallet with provided credentials', async () => {
+      const email = 'alice@mail.com'
+      const password = 'qwe123'
+
+      wrapper.setMethods({
+        isUserExist: sinon.stub().resolves(true)
+      })
+      wrapper.setData({
+        form: {
+          email,
+          password
+        }
+      })
+
+      await wrapper.vm.submit()
+      // Vue will dispatch the action with the store object as a first param
+      // so it's much easier to check only needed param instead of doing
+      // spy.withArgs()
+      expect(spies.loadWallet.calledOnce).to.be.true
+      expect(spies.loadWallet.args[0][1]).to.deep.equal({ email, password })
```

Comment:
I remember a way to partially test provided params instead of full match in sinon

Reply:
resolved, fixed